tests/archive.rs

```diff
@@ -51,3 +51,44 @@ fn archives_are_disabled(tmpdir: TempDir, port: u16) -> Result<(), Error> {
 
     Ok(())
 }
+
+#[rstest]
```

Comment:
If you wanna do something cool here, you could use rstest for test parameterization to test `-p` and `-g` with very little test code overhead. Want to give that a try?

Reply:
Actually, on second thought, I want to more properly clean up these tests anyhow. I'll just do that after merging.

Reply:
I can try in maybe another PR?

Reply:
Sure, if you like, that'd be much appreciated. So what I'm thinking here: Have a single parametrized test in `tests/archive.rs` that would test for all possible combinations of compression flags whether the path is reachable whether the links are being rendered. Thanks to rstest, we'd end up with a bunch of separately generated distinct tests. I think that'd be rather clean and make quite a bit of sense. Wanna give it a shot?

Take a look at: https://docs.rs/rstest/0.7.0/rstest/attr.rstest.html#test-parametrized-cases